modules/light-clients/08-wasm/types/vm.go

```diff
@@ -42,7 +46,12 @@ func callContract(ctx sdk.Context, clientStore storetypes.KVStore, codeHash []by
 	sdkGasMeter := ctx.GasMeter()
 	multipliedGasMeter := NewMultipliedGasMeter(sdkGasMeter, VMGasRegister)
 	gasLimit := VMGasRegister.runtimeGasForContract(ctx)
-	env := getEnv(ctx)
+
+	clientID, err := getClientID(clientStore)
+	if err != nil {
+		return nil, errorsmod.Wrapf(err, "failed to retrieve clientID for wasm contract call")
+	}
+	env := getEnv(ctx, clientID)
```

Comment:
I didn't move this logic inside `getEnv` [because `NewEnv` in wasmd also expects a contractAddr](https://github.com/CosmWasm/wasmd/blob/2eee755048f34295d581c44a56227cb932043367/x/wasm/types/types.go#L270-L295)

Reply:
but would it make more sense to expect a `contrAddr` in `NewEnv` as opposed to `getEnv`?

Reply:
I treated `NewEnv` and `getEnv` as the same. Both of them expect a `contractAddr` now. I think this comment I created is redundant and confusing, sorry about that. Feel free to resolve this comment as I don't think it adds much value.